### PR TITLE
Made NS 2.0 adjustments

### PR DESCRIPTION
**Pull request by davidfowl**
Reduce the netstandard 2.0 dependency graph

PS: I'm not sure why **System.Security.Cryptography.Cng** was marked as PrivateAssets="All", that seems broken. I didn't run tests but I installed the packages build to verify the graph was indeed smaller. Hopefully your CI runs tests to verify I didn't break anything 😄

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj [hunk 1]
```diff
@@ -62,6 +62,11 @@
     <PackageReference Include="System.Threading" Version="4.3.0-*" />
     <PackageReference Include="System.Xml.XmlDocument" Version="4.3.0" />
   </ItemGroup>
+  
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' ">
+    <PackageReference Include="Microsoft.CSharp" Version="4.5.0" />
```
Comment (GeoK) [inline src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj]: Not really sure why this package reference is needed. Could you please explain?

**Comment by GeoK** [comment]
Hi @davidfowl, thank you for the PR.
Our build complains that there is a package downgrade for System.Security.Cryptography.Cng (4.5.0 to 4.3.0) in a couple of our test projects.

As these references are actually not needed, please feel free to remove them.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/72f03a9c62215688b759ea180263db8b498280ea/test/Microsoft.IdentityModel.Tokens.Tests/Microsoft.IdentityModel.Tokens.Tests.csproj#L38-L41

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/72f03a9c62215688b759ea180263db8b498280ea/test/Microsoft.IdentityModel.TestUtils/Microsoft.IdentityModel.TestUtils.csproj#L34-L36

File: src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj (hunk 1 above)
Comment (davidfowl) [reply src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj]: Are you using dynamic anywhere?

Comment (GeoK) [reply src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj]: Oh, got it. Thank you.

**Comment by davidfowl** [comment]
I’ll send an update

File: src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj (hunk 1 above)
Comment (davidfowl) [reply src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj]: You should just remove the use of dynamic though.

**Comment by davidfowl** [comment]
Updated a couple of tests. I can't run build.cmd. I tried to run some of the tests but some ran and some failed with weird binding errors. There's no PR checks enabled on this repo so it makes it hard to know if it passes the build.

**Comment by GeoK** [comment]
Thank you for the update. I'm guessing that strong name validation fails as the assemblies are delay signed. We have two scripts inside the build folder to bypass strong name validation.

We've just enabled PR validation.

**Comment by davidfowl** [comment]
@GeoK is this good to go?

**Comment by brentschmaltz** [comment]
:shipit:  @davidfowl what errors are you getting from build.cmd?

**Pull request merged**